src/devices/BuildHat/Brick.cs

```diff
@@ -0,0 +1,1564 @@
+            // Set continuous reading
+            SelectCombiModesAndRead(port, new int[] { 1, 2, 3 }, false);
+            // Ramp uses first param as initial position, second as target, third is how long, foruth is always 0
+            PortWrite($"port {(byte)port} ; pid {(byte)port} 0 1 s4 0.0027777778 0 5 0 .1 3 ; set ramp {actualPositionDouble.ToString(CultureInfo.InvariantCulture)} {newPosition.ToString(CultureInfo.InvariantCulture)} {duration.ToString(CultureInfo.InvariantCulture)} 0\r");
+            if (blocking)
+            {
+                Thread.Sleep((int)(duration * 1000));
+            }
+        }
+
+        /// <summary>
+        /// Run the motor to an absolute position.
+        /// </summary>
+        /// <param name="port">The port.</param>
+        /// <param name="targetPosition">The target angle from -180 to +180.</param>
+        /// <param name="speed">The speed from - 100 to 100.</param>
+        /// <param name="blocking">True to block the function and wait for the execution.</param>
+        /// <exception cref="ArgumentException">Not a motor or not an active motor.</exception>
+        public void MoveMotorToPosition(SensorPort port, int targetPosition, int speed, bool blocking = false)
+        {
+            if (!IsMotor(_sensorType[(byte)port]))
+            {
+                throw new ArgumentException("Not a motor connected");
+            }
+
+            if (!IsActiveSensor(_sensorType[(byte)port]))
+            {
+                throw new ArgumentException("Not an active motor connected");
+            }
+
+            if (speed == 0)
+            {
+                throw new ArgumentException("Speed can't be 0");
+            }
+
+            speed = speed < -100 ? -100 : speed > 100 ? 100 : speed;
+            ActiveMotor motor = (ActiveMotor)_elements[(byte)port];
+            // We need both the relative
+            int actualPosition = motor.Position;
+            double actualPositionDouble = actualPosition / 360.0;
+            double newPosition = (targetPosition - actualPositionDouble) / 360.0;
+            double duration = Math.Abs(newPosition - actualPositionDouble) / (speed * 0.05 / motor.PowerLimit);
+            // Set continuous reading
+            SelectCombiModesAndRead(port, new int[] { 1, 2, 3 }, false);
+            // Ramp uses first param as initial position, second as target, third is how long, foruth is always 0
+            PortWrite($"port {(byte)port} ; pid {(byte)port} 0 1 s4 0.0027777778 0 5 0 .1 3 ; set ramp {actualPositionDouble.ToString(CultureInfo.InvariantCulture)} {newPosition.ToString(CultureInfo.InvariantCulture)} {duration.ToString(CultureInfo.InvariantCulture)} 0\r");
+            if (blocking)
+            {
+                while (!((motor.Position / 360.0 < newPosition + 2.028) && (motor.Position / 360.0 > newPosition - 2.028)))
+                {
+                    Thread.Sleep(5);
+                }
+            }
```

Comment:
Suggestion: Make these blocking functions async or return a wait handle.

Reply:
I was thinking of this and made the choice to have it without async. There are 2 reasons, first, the target audience for this hat are beginners and second it's for the port then to nanoFramework. I can add an async one if you think it's needed

Reply:
That makes sense. Fine with me. 

But another thing I just see: You should not read out `motor.Position` twice in the condition, as this could result in undefined behavior, because the value might change between the two calls. Additionally, it seems that `2.028` should be a constant.